pkg/reconciler/workload/apiexport/workload_apiexport_reconcile.go

```diff
@@ -145,15 +125,17 @@ func (r *schemaReconciler) reconcile(ctx context.Context, export *apisv1alpha1.A
 		if equality.Semantic.DeepEqual(&existingSchema.Spec, &newSchema.Spec) {
 			// nothing to do
 			upToDate.Insert(resourceGroup)
-			schemasByResourceGroup[resourceGroup] = existingSchema
 		}
 	}
 
 	// create missing or outdated schemas
 	outdatedOrMissing := expectedResourceGroups.Difference(upToDate)
 	for _, resourceGroup := range outdatedOrMissing.List() {
 		logger.WithValues("schema", resourceGroup).V(2).Info("missing or outdated schema on APIExport, adding")
-		resource := resourcesByResourceGroup[resourceGroup]
+		resource, ok := resourcesByResourceGroup[resourceGroup]
```

Comment:
I'm not sure about the impact of this line (continuing if the missing resource is not found in the NegotiatedAPIresources).
Will the APIResourceSchema be removed from the APIExport finally because it was not found in the NegotiatedAPIResources ?
What if this resource later exists in the downstream cluster ? the schema would not be added to the APIExport, and we would have lost one of the resources initially added in the APIExport by the `workload sync` CLI command.

I might be missing something but the logic is not clear to me.

Reply:
it won't, the APIResourceSchema reference in the  APIExport  is only replaced by the updated one. I agree the logic here is not quite clear. Need to do a refactor